xmodule/discussion_block.py

```diff
@@ -73,13 +73,6 @@ class DiscussionXBlock(XBlock, StudioEditableXBlockMixin, XmlParserMixin):  # li
 
     @property
     def course_key(self):
-        """
-        :return: int course id
-
-        NB: The goal is to move this XBlock out of edx-platform, and so we use
-        scope_ids.usage_id instead of runtime.course_id so that the code will
-        continue to work with workbench-based testing.
-        """
         return getattr(self.scope_ids.usage_id, 'course_key', None)
```

Comment:
It's interesting that DiscussionBlock uses this pattern. Is it possible for calls to `scope_ids.usage_id.context_key` to fail if it's an Old Mongo course? We are actively removing most of Old Mongo right now, but parts of it will need to continue to work (like the root CourseBlock).

Reply:
I was worried that with Old Mongo, the real course key with run information in it wouldn't get added, since runs aren't derivable from the old-style Usage Keys. I thought that maybe we'd get scenarios were we didn't call `map_into_course` properly. But I forgot that Old Mongo has that hacky [`fill_in_run`](https://github.com/openedx/edx-platform/blob/d9f54b9f460bed0001e999dac8884d3b8e8a949f/xmodule/modulestore/mongo/base.py#L614) code for that, which gets called during [`load_item`](https://github.com/openedx/edx-platform/blob/a389a9ff10fd06bd053107fc6a4f19c7d24f7233/xmodule/modulestore/mongo/base.py#L226-L231).